src/model/topology.cpp

```diff
@@ -194,12 +194,14 @@ void Topology::Specs::ParseAccelergyERT(config::CompoundConfigNode ert)
       for (unsigned i = 0; i < NumNetworks(); i++) {
           // only check the user-defined networks
           auto networkSpec = GetNetwork(i);
-          if (networkSpec->Type() == "SimpleMulticast" && networkSpec->name == componentName){
-            assert(false); // FIXME: why is the name of a network class exposed in this file?
-            //if (std::static_pointer_cast<SimpleMulticastNetwork::Specs>(networkSpec)){
-            //  // std::cout << "simple multicast component identified: " << componentName << std::endl;
-            //  std::static_pointer_cast<SimpleMulticastNetwork::Specs>(networkSpec)->accelergyERT = componentERT;
-            //}
+          if (networkSpec->SupportAccelergyTables() && networkSpec->name == componentName){
+            // assert(false); // FIXME: why is the name of a network class exposed in this file?
+            // Only simplemulticast network support ERT parsing,
+            // Instead using the exact name, added an interface function for the check
+            if (std::static_pointer_cast<SimpleMulticastNetwork::Specs>(networkSpec)){
+              // std::cout << "NoC that supports Acceleragy tables identified: " << componentName << std::endl;
+              std::static_pointer_cast<SimpleMulticastNetwork::Specs>(networkSpec)->accelergyERT = componentERT;
```

Comment:
I liked your SupportAccelergyTables() abstraction above, but this line is bad. It won't scale to anything other than SimpleMulticastNetwork. You could add another virtual method, e.g., SetAccelergyERT(). Default implementation would assert false, while an overridden implementation would do something with it.